### ISSUE TITLE
Improved insertion sort

### ISSUE THREAD

**Issue by hasinaxp** [dscgcelt/Competetive-Coding-Repo#15]
To improve insertion sort you can reduce the search time of finding the place to insert . One approach will be to use binary search instead of linear search.😉

**Comment by jddeep** [comment]
Yes :+1: Anyone wants to work on this can do so. Make the changes in the present .cpp file in the PR.

**Comment by Sayan-dev** [comment]
lets do it

**Comment by tarmas99** [comment]
Try to validate your proposal by creating a large list of  say 10^7 randomly generated elements, and compare the timing. 
See which type of searching technique improves the time. 
No need to do complexity analysis now, get a understanding of it by looking at the amount of time it takes.

**Comment by tarmas99** [comment]
@hasinaxp @jddeep @Sayan-dev if not wrong, the main condition for binary search is to have the array of interest to be sorted before hand.
